### PR TITLE
fix: Update defect of of wrong resource attribute of "container.id"

### PR DESCRIPTION
**Pull request by liurui-1**
## Which problem is this PR solving?

As described in "https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1677", the way to get resource attribute of "container.id" is wrong.

## Short description of the changes

I made the changes following the code of OTel Java SDK which works well.
Here is the code I followed:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ContainerResource.java#L54-L60

I have tested the code working well in my Kubernetes environments.

### CODE REVIEW

**Comment by pichlermarc** [comment]
Thanks for fixing this! :slightly_smiling_face:
Looks like the tests still need some attention as their current test data is not very realistic, so the new logic drops them. 

Feel free to take the test-data from https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/0f1199620db05a3739cd3bae7e0c408a9bc0d65b/instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/CgroupV1ContainerIdExtractorTest.java#L82-L113 and adapt the tests here to use those. :slightly_smiling_face:

**Comment by liurui-1** [comment]
Made some slight change. Please review again.

**Comment by liurui-1** [comment]
Hello, I updated the UT code. Please review again.

**Comment by liurui-1** [comment]
Hello, I updated the UT code. Please review again.

**Comment by liurui-1** [comment]
@pichlermarc, I updated code according to the lint message. It seems the lint result does not show up automatically. Please help review.

File: detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts [hunk 1]
```diff
@@ -57,9 +57,29 @@ export class ContainerDetector implements Detector {
...
     );
     const splitData = rawData.trim().split('\n');
-    for (const str of splitData) {
-      if (str.length >= this.CONTAINER_ID_LENGTH) {
-        return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+    for (const line of splitData) {
+      const lastSlashIdx = line.lastIndexOf('/');
+      if (lastSlashIdx === -1){
+        continue;
+      }
```
Comment (pichlermarc) [inline detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts:64]: looks like there was just some missing whitespace ` `
```suggestion
      if (lastSlashIdx === -1) {
        continue;
      }
```

**Pull request merged**